src/memray/reporters/flamegraph.py

```diff
@@ -1,22 +1,35 @@
 import html
 import linecache
 import sys
+from itertools import tee
+from itertools import zip_longest
 from typing import Any
 from typing import Dict
 from typing import Iterable
 from typing import Iterator
 from typing import TextIO
+from typing import Tuple
+from typing import TypeVar
 
 from memray import AllocationRecord
 from memray import MemorySnapshot
 from memray import Metadata
 from memray.reporters.frame_tools import StackFrame
 from memray.reporters.frame_tools import is_cpython_internal
+from memray.reporters.frame_tools import is_frame_from_import_system
 from memray.reporters.frame_tools import is_frame_interesting
 from memray.reporters.templates import render_report
 
 MAX_STACKS = int(sys.getrecursionlimit() // 2.5)
 
+T = TypeVar("T")
+
+
+def pairwise(iterable: Iterator[T]) -> Iterable[Tuple[T, T]]:
+    a, b = tee(iterable)
+    next(b, None)
+    return zip_longest(a, b)
```

Comment:
Let's not call this `pairwise`, since it's slightly different from `itertools.pairwise` (`zip_longest` vs `zip`).

I almost don't care what it's called, as long as it's anything but that. I originally thought this was just a backport of `itertools.pairwise`, but it's not, and that's gonna confuse future us.